### DISCUSSION TITLE
Why are there two ThreadFactoryProvider?

### DISCUSSION THREAD

**Discussion by inomp3042** [redis/lettuce#2052] [Q&A]
I don't understand that there are two ThreadFactoryProvider interfaces. Is there a reason?

1. DefaultEventLoopGroupProvider.ThreadFactoryProvider https://github.com/lettuce-io/lettuce-core/blob/640be7c872c7266cd86259326acd02cb696994f3/src/main/java/io/lettuce/core/resource/DefaultEventLoopGroupProvider.java#L341-L352
2. ThreadFactoryProvider https://github.com/lettuce-io/lettuce-core/blob/640be7c872c7266cd86259326acd02cb696994f3/src/main/java/io/lettuce/core/resource/ThreadFactoryProvider.java#L27-L37

If there aren't any reasons, may I delete `DefaultEventLoopGroupProvider.ThreadFactoryProvider`?

**Comment by mp911de** [comment]
We can remove `DefaultEventLoopGroupProvider.ThreadFactoryProvider` with Lettuce 7, the next major revision. The variant in the `resource` package was introduced as a general utility to allow configuration of `HashedWheelTimer` threads. The inner class design of `DefaultEventLoopGroupProvider` doesn't really fit a general usage pattern. We left the provider in `DefaultEventLoopGroupProvider` for compatibility reasons in where it is.
